dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/model/MethodMetric.java

```diff
@@ -38,10 +41,17 @@ public class MethodMetric extends ServiceKeyMetric {
 
     public MethodMetric(ApplicationModel applicationModel, Invocation invocation) {
         super(applicationModel, MetricsSupport.getInterfaceName(invocation));
-        this.methodName = RpcUtils.getMethodName(invocation);
         this.side = MetricsSupport.getSide(invocation);
         this.group = MetricsSupport.getGroup(invocation);
         this.version = MetricsSupport.getVersion(invocation);
+        this.methodName = isServiceLevel(applicationModel) ? null : RpcUtils.getMethodName(invocation);
```

Comment:
Should pre-check for `isServiceLevel`. Prevent cal it for each invocation.

Reply:
Thank you, this is a good suggestion